content/flux/v0.x/join-data/_index.md

```diff
@@ -0,0 +1,143 @@
+  {{< children type="anchored-list" filterOut="Troubleshoot join operations" >}}
+- [Troubleshoot join operations](#troubleshoot-join-operations)
+
+## How join functions work
+
+All functions in the `join` package join _two_ streams of tables together based 
+on common values in each input stream.
+
+- [Input streams](#input-streams)
+- [Join predicate function (on)](#join-predicate-function-on)
+- [Join output function (as)](#join-output-function-as)
+
+### Input streams
+
+Each input stream is assigned to the `left` or `right` parameter.
+Input streams can be defined from any valid data source.
+For more information, see:
+
+- [Query data sources](/flux/v0.x/query-data/)
+- Define ad hoc tables with [`array.from()`](/flux/v0.x/stdlib/array/from/)
+
+#### Data requirements
+
+To join two streams of data with the `join` package, each stream must have:
+
+- **One or more columns with common values to join on**.  
+  Columns do not need identical labels, but they do need to have comparable values.
+- **Identical [group keys](/flux/v0.x/get-started/data-model/#group-key)**.  
+  Functions in the `join` package use group keys to quickly determine what tables
+  from each input stream should be paired and evaluated for the join operation.  
+  _Both input streams must have the same group key._
+  This likely requires using [`group()`](/flux/v0.x/stdlib/universe/group/)
+  to regroup each input stream before joining them together.
+
+  {{% note %}}
+Only tables with the same [group key instance](/flux/v0.x/get-started/data-model/#example-group-key-instances)
+are joined.
+  {{% /note %}}
+
+### Join predicate function (on)
+
+Each function in the `join` package requires the `on` parameter which is the
+**join predicate**—a [predicate function](/flux/v0.x/get-started/syntax-basics/#predicate-functions)
+that compares values from each input stream (represented by `l` (left) and `r` (right))
+and returns `true` or `false`.
+Rows that return `true` when compared are joined.
+Rows that return `false` when compared are not.
+
+```js
+(l, r) => l.column == r.column
+```
+
+{{% note %}}
+Only tables that share the same group key instance are evaluated by the join predicate.
+Because of this, **both input streams must have the same [group key](/flux/v0.x/get-started/data-model/#group-key)**.
+{{% /note %}}
+
+### Join output function (as)
+
+All functions in the `join` package _(except [`join.time()`](/flux/v0.x/stdlib/join/time/))_
+require the `as` parameter which defines the output schema of the join.
+The `as` parameter is a function that returns a a new record using values from
+joined rows–left (`l`) and right (`r`).
+
+```js
+(l, r) => ({l with name: r.name, location: r.location})
+```
+
+For left, right, and full outer joins, the `as` functions must account for _null_
+values that may occur in each input stream. For more information, see the
+[join types documentation](#perform-join-operations) below.
```

Comment:
shall we link to the procedures directly? it feels a bit duplicate to have the perform join operations on both this page and the individual procedures. As best practice, think we want each procedure to stand on its own outside, for folks who don't want to read conceptual content